stores/metrics.go

```diff
@@ -348,9 +389,42 @@ func (s *SQLStore) contractMetrics(ctx context.Context, start time.Time, n uint6
 	}
 
 	var metrics []dbContractMetric
-	err := s.findPeriods(tx, &metrics, start, n, interval)
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch contract metrics: %w", err)
+	if opts.ContractID == (types.FileContractID{}) && opts.HostKey == (types.PublicKey{}) {
+		// if neither contract nor host filters were set, we return the
+		// aggregate spending for each period
+		var metricsWithPeriod []struct {
+			Metric dbContractMetric `gorm:"embedded"`
+			Period int64
+		}
+		currentPeriod := int64(math.MinInt64)
+		err := tx.Raw(`
+			SELECT * FROM contracts
```

Comment:
maybe reformat this raw query block to remove leading tabs/spaces and make the inner select more clear

Reply:
I added some indentation to make it look nicer but I'm not sure how to get rid of tabs while at the same time making it more clear by itself and with the surrounding code.